geo-traits/src/to_geo.rs

```diff
@@ -0,0 +1,163 @@
+//! Convert structs that implement geo-traits to [geo-types] objects.
+
+use geo_types::{
+    Coord, CoordNum, Geometry, GeometryCollection, Line, LineString, MultiLineString, MultiPoint,
+    MultiPolygon, Point, Polygon, Rect, Triangle,
+};
+
+use crate::{
+    CoordTrait, GeometryCollectionTrait, GeometryTrait, GeometryType, LineStringTrait, LineTrait,
+    MultiLineStringTrait, MultiPointTrait, MultiPolygonTrait, PointTrait, PolygonTrait, RectTrait,
+    TriangleTrait,
+};
+
+/// Convert any coordinate to a [`Coord`].
+///
+/// Only the first two dimensions will be kept.
+pub fn coord_to_geo<T: CoordNum>(coord: &impl CoordTrait<T = T>) -> Coord<T> {
+    Coord {
+        x: coord.x(),
+        y: coord.y(),
+    }
+}
+
+/// Convert any Point to a [`Point`].
+///
+/// Only the first two dimensions will be kept.
+pub fn point_to_geo<T: CoordNum>(point: &impl PointTrait<T = T>) -> Point<T> {
+    if let Some(coord) = point.coord() {
+        Point(coord_to_geo(&coord))
+    } else {
+        panic!("converting empty point to geo not implemented")
```

Comment:
I agree this panic is unfortunate and not ideal.

> In this case, how about returning an `Option<Point<T>>`. Note it would cascade to the `to_geometry` function as well.

I'm open to this for `to_point` but I think it seems rather bad for `to_geometry`. 

Perhaps the best way is to have panicking and non-panicking variants? 🤷‍♂️

Reply:
Yeah, panicking and `try_*` non-packing variants seems like a good middleground.

Reply:
I added panicking and non-panicking variants